### PR TITLE
Add an option to use GroupNorm instead of BatchNorm in Conformer model

### PR DESCRIPTION
**Pull request by xiaohui-zhang**
gpurun pytest test/torchaudio_unittest/models/conformer
git add ============================= test session starts ==============================
platform linux -- Python 3.7.0, pytest-7.1.1, pluggy-1.0.0
rootdir: /data/home/xiaohuizhang/audio_dev2
collected 4 items

test/torchaudio_unittest/models/conformer/conformer_cpu_test.py ..       [ 50%]
test/torchaudio_unittest/models/conformer/conformer_gpu_test.py ..       [100%]

============================== 4 passed in 10.36s ==============================

### CODE REVIEW

File: torchaudio/models/conformer.py [hunk 1]
```diff
@@ -131,6 +134,7 @@ def __init__(
         num_attention_heads: int,
         depthwise_conv_kernel_size: int,
         dropout: float = 0.0,
+        use_group_norm: bool = False,
```
Comment (hwangjeff) [inline torchaudio/models/conformer.py:137]: can you add a corresponding entry to the docstring?

File: torchaudio/models/conformer.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def __init__(
         num_channels: int,
         depthwise_kernel_size: int,
         bias: bool = False,
+        use_group_norm: bool = False,
```
Comment (hwangjeff) [inline torchaudio/models/conformer.py:34]: and similarly here?

also noticed that `dropout` is missing a docstring...we can address separately

File: torchaudio/models/conformer.py [hunk 3]
```diff
@@ -22,16 +22,19 @@ class _ConvolutionModule(torch.nn.Module):
...
     """
 
     def __init__(
         self,
         input_dim: int,
         num_channels: int,
         depthwise_kernel_size: int,
-        bias: bool = False,
         dropout: float = 0.0,
+        bias: bool = False,
```
Comment (mthrok) [inline torchaudio/models/conformer.py:36]: Moving `bias` parameter is BC-breaking. Is this intended?

Comment (xiaohui-zhang) [reply torchaudio/models/conformer.py:36]: sorry what does it mean by BC-breaking? I just wanted to gather bool-type inputs together, after int/float inputs

Comment (mthrok) [reply torchaudio/models/conformer.py:36]: Changing the order of function arguments breaks backward compatibility.

Say, a library, FooLibrary of v1 has a function like the following

```python
foo(a: int, b: str)
```

and in my project code I use it as

```python
result = foo(3, "foo")
```

Now, if the library code is changed like the following in v2 or something,

```python
foo(b: str, a: int)
```

Then simply upgrading my environment from FooLibrary v1 to v2, my project code stops working, unless I change my own code. The kind of changes that force client code change, is backward-incompatible, and it has to be communicated with users (deprecation). In PyTorch, we take at least two-release cycles for deprecating a public API.

Comment (mthrok) [reply torchaudio/models/conformer.py:36]: however the said class is not a public API, rather an internal one, so I guess it's kinda okay. Although I would say it's unnecessary if it's for the sake of cosmetics. @hwangjeff thoughts?

File: torchaudio/models/conformer.py [hunk 4]
```diff
@@ -122,6 +127,8 @@ class ConformerLayer(torch.nn.Module):
         num_attention_heads (int): number of attention heads.
         depthwise_conv_kernel_size (int): kernel size of depthwise convolution layer.
         dropout (float, optional): dropout probability. (Default: 0.0)
+        use_group_norm (bool, optional): use GroupNorm rather than BatchNorm in ConvolutionModule. (Default: False)
+        convolution_first (bool, optional): apply the convolution module ahead of the attention module. (Default: False)
```
Comment (hwangjeff) [inline torchaudio/models/conformer.py:131]:
```suggestion
        convolution_first (bool, optional): apply the convolution module ahead of the attention module. (Default: ``False``)
```

File: torchaudio/models/conformer.py (hunk 3 above)
Comment (xiaohui-zhang) [reply torchaudio/models/conformer.py:36]: it's an internal function so it should be ok to change, though I agree it's really a cosmetics change...

**Review by hwangjeff — APPROVED** [review]
lg
